core-plugins/src/e2e-test/features/deduplicate/Deduplicate_Macro.feature

```diff
@@ -0,0 +1,60 @@
+    Then Verify the Output Schema matches the Expected Schema: "deduplicateOutputSchema"
+    Then Validate "File" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "Deduplicate"
+    Then Click on the Macro button of Property: "deduplicateUniqueFields" and set the value to: "deduplicateUniqueFields"
+    Then Click on the Macro button of Property: "filterOperation" and set the value to: "deduplicateFilterOperation"
+    Then Click on the Macro button of Property: "deduplicateNumPartitions" and set the value to: "deduplicateNumberOfPartitions"
+    Then Validate "Deduplicate" plugin properties
+    Then Close the Plugin Properties page
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "File" from the plugins list as: "Sink"
+    Then Connect plugins: "Deduplicate" and "File2" to establish connection
+    Then Navigate to the properties page of plugin: "File2"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "filePluginOutputFolder"
+    Then Replace input plugin property: "pathSuffix" with value: "yyyy-MM-dd-HH-mm-ss"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Click plugin property: "writeHeader"
+    Then Validate "File" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Enter runtime argument value "deduplicateUniqueFields" for key "deduplicateUniqueFields"
+    Then Enter runtime argument value "deduplicateFilterOperation" for key "deduplicateFilterOperation"
+    Then Enter runtime argument value "deduplicateNumberOfPartitions" for key "deduplicateNumberOfPartitions"
+    Then Run the preview of pipeline with runtime arguments
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "deduplicateUniqueFields" for key "deduplicateUniqueFields"
+    Then Enter runtime argument value "deduplicateFilterOperation" for key "deduplicateFilterOperation"
+    Then Enter runtime argument value "deduplicateNumberOfPartitions" for key "deduplicateNumberOfPartitions"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count of deduplicate is equal to IN record count of sink
+#    Enable below step once issue fixed (PLUGIN #1310)
+#    Then Validate output file generated by file sink plugin is equal to expected output file {string}
```

Comment:
Validation logic should be indepedent of the number of part files. You should read data from all the part files and verify that the data read matches the expectation

Reply:
Added the validation step.